### ISSUE TITLE
allow kops to specify tenancy parameters

### ISSUE THREAD

**Issue by hollowimage** [kubernetes/kops#1756]
add
`--node-tenancy [tenancy]` and `--master-tenancy [tenancy]` so I can pass dedicated or host to kops and have it create launch configurations without default to inheriting these values from the target VPC if you are using a shared vpc configuration.

**Comment by chrislovecnm** [comment]
Can you add more context?  `kops` provisions all instances.

**Comment by hollowimage** [comment]
So we're operating under HIPAA guidelines, and one of such requirements is to have instances be tenancy=dedicated.

in aws you can either achieve that by either setting vpc type to dedicated (which will force all instances created inside it to be of dedicated type) or set the instance on a flag directly when creating it into a non-dedictated vpc.

kops creates all instances but it does not specify tenancy, and creates launch configuration for master and nodes, which inherits from VPC, by default.

**Comment by hollowimage** [comment]
Is this still being looked at? I think we're on 1.5.3 now. I could help if needed.

**Comment by wmbutler** [comment]
It's definitely a requirement under the AWS HIPPA guidelines so it would be a nice add. Here's a snippet of a cloudformation template so you can see the syntax.

```json
"LaunchConfiguration": {
       "Type": "AWS::AutoScaling::LaunchConfiguration",
       "Properties": {
          "ImageId": "ami-71852f64",
          "InstanceType": "c3.2xlarge",
          "PlacementTenancy" : "dedicated",
          "EbsOptimized" : "true",
          "AssociatePublicIpAddress" : "true",
          "SecurityGroups" : ["sg-83cc89e7"],
            }
          }
       }
    },
```

Would be great to add it to the node here:

```yml
apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: "2017-03-20T16:18:43Z"
  labels:
    kops.k8s.io/cluster: uswest2.cluster.cavo.io
  name: nodes
spec:
  associatePublicIp: true
  image: ami-72852f64
  machineType: c3.2xlarge
  placementTenancy: dedicated
  maxSize: 2
  minSize: 2
  role: Node
  subnets:
  - us-west-2a
```

**Comment by chrislovecnm** [comment]
Closing this as we have it supported now.  I would need to check the code, but this should be supported in 1.6 alpha releases.  If not it will be included in the 1.6 release.
